addOns/ascanrulesBeta/src/test/java/org/zaproxy/zap/extension/ascanrulesBeta/SourceCodeDisclosureFileInclusionScanRuleUnitTest.java

```diff
@@ -85,4 +101,141 @@ void shouldReturnExpectedExampleAlert() {
         assertThat(alert.getRisk(), is(equalTo(Alert.RISK_HIGH)));
         assertThat(alert.getConfidence(), is(equalTo(Alert.CONFIDENCE_MEDIUM)));
     }
+
+    @Test
+    void shouldSkipUrlParams() {
+        // Given
+        HttpMessage msg = createMessage("/test/");
+        NameValuePair nvp = new NameValuePair(NameValuePair.TYPE_URL_PATH, "inc", "", 0);
+        // When
+        rule.scan(msg, nvp);
+        // Then
+        assertThat(countMessagesSent, is(equalTo(0)));
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"", "/"})
+    void shouldSkipIfNoOrEmptyPath(String path) {
+        // Given
+        HttpMessage msg = createMessage(path);
+        rule.init(msg, parent);
+        // When
+        rule.scan();
+        // Then
+        assertThat(countMessagesSent, is(equalTo(0)));
+    }
+
+    @Test
+    void shouldNotContinueIfRandomBodyIsEmpty() throws IOException {
+        // Given
+        String path = "/shouldNotContinueIfRandomBodyIsEmpty";
+        nano.addHandler(new FiHandler(path, Response.Status.OK, ""));
+        HttpMessage msg = this.getHttpMessage(path + "?inc=bar");
+        rule.init(msg, parent);
+        // When
+        rule.scan();
+        // Then
+        assertThat(httpMessagesSent.size(), is(equalTo(2)));
+        HttpMessage m = httpMessagesSent.get(1);
+        assertThat(m.getResponseBody().toString(), is(equalTo("")));
+    }
+
+    @Test
+    void shouldNotContinueIfRandomBodyIsTooSimilar() throws IOException {
+        // Given
+        String path = "/shouldNotContinueIfRandomBodyIsTooSimilar";
+        nano.addHandler(new FiHandler(path, Response.Status.OK, DEFAULT_RESPONSE_STRING));
+        HttpMessage msg = this.getHttpMessage("GET", path + "?inc=bar", DEFAULT_RESPONSE_STRING);
+        rule.init(msg, parent);
+        // When
+        rule.scan();
+        // Then
+        assertThat(httpMessagesSent.size(), is(equalTo(2)));
+        HttpMessage m = httpMessagesSent.get(1);
+        assertThat(m.getResponseBody().toString(), is(equalTo(DEFAULT_RESPONSE_STRING)));
+    }
+
+    @Test
+    void shouldAlertWhenConditionsRightAndPhpExtensionMatchesContent() throws IOException {
```

Comment:
Should be paremeterized or renamed to be generic (same for the following).

Reply:
What would be parameterized in the next one, no extension is a single case